File: src/main/java/org/bakkes/game/model/entity/Character.java
package org.bakkes.game.model.entity;

import org.bakkes.game.model.map.Direction;
import org.bakkes.game.model.map.Tile;

public abstract class Character extends OverworldEntity {

	protected int facing = Direction.SOUTH;
	public int getFacing() {
		return facing;
	}
	public void setFacing(final int direction){
		facing = direction;
	}

	private boolean isWalking = false;
	public void onEnterNewTile(){
		isWalking = true;
	}
	public void onFinishedWalking(){
		isWalking = false;
	}
	public boolean isWalking(){
		return isWalking;
	}
	/**
	 * used a lot, anoying to do
	 * @return
	 */
	public Tile getDirectionTile(){
        final int facing = getFacing();
        final Tile result = new Tile();
        switch(facing){
        	case Direction.NORTH:
        		result.top-=1;
        		break;
        	case Direction.SOUTH:
        		result.top+=1;
        		break;
        	case Direction.EAST:
        		result.left+=1;
        		break;
        	case Direction.WEST:
        		result.left-=1;
        		break;
        }
        return result;
	}
}

File: src/main/scala/org/bakkes/game/model/entity/Character.scala
package org.bakkes.game.model.entity

import org.bakkes.game.model.map.Direction
import org.bakkes.game.model.map.Direction._
import org.bakkes.game.model.map.Tile

import scala.beans.BeanProperty

abstract class Character(
	@BeanProperty
	var facing: Direction
) extends OverworldEntity {

	def this(){
		this(South)
	}
	private var _isWalking: Boolean = false

	def onEnterNewTile = _isWalking = true
	def onFinishedWalking = _isWalking = false

	def isWalking: Boolean = _isWalking

	/**
	 * used a lot, anoying to do
	 * @return
	 */
	def getDirectionTile: Tile = {
		var result = new Tile
		getFacing match {
			case North => result -= Tile(0,1)
			case South => result += Tile(0,1)
			case East => result += Tile(1,0)
			case West => result -= Tile(1,0)
		}
		return result
	}
}